### PR TITLE
remove DetachedEagleGPT model and handle all offline mode in the _DynamicEagleGPTModel

### PR DESCRIPTION
**Pull request by yeyu-nvidia**
## What does this PR do?

**Type of change:** refactor

**Overview:**
Following HF implementation, remove the dedicate Detached Eagle class and handle offline case in the same class as online mode.

## Usage

```python
# Add a code snippet demonstrating how to use this
```

## Testing
Tested with both online and offline Qwen3 30B for EAGLE-3 training.

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Summary by CodeRabbit

- New Features
  - Added an offline EAGLE execution path with sequence parallelism disabled.
  - Supported additional EAGLE configuration options (self-logit distillation, freeze base model, report accuracy, reuse base decoder, loss decay, architecture config).
  - Improved label/logit shape handling for offline training and multi-step losses.

- Refactor
  - Consolidated to a single unified registry for EAGLE models; offline registry removed.
  - Removed detached/offline wrappers and related registrations.
  - Streamlined conversion to always use the unified registry and updated forward paths for online/offline branching.

### CODE REVIEW

**Comment by ChenhanYu** [comment]
Need to run the regression test to see if any impact to the megatron online training.

File: modelopt/torch/speculative/plugins/megatron_eagle.py [hunk 1]
```diff
@@ -1305,7 +1341,7 @@ def forward(
             packed_seq_params=packed_seq_params,
             **(extra_block_kwargs or {}),
         )
-        eagle_logits_1 = eagle_logits_2x[labels.shape[1] :, :, :]
+        eagle_logits_1 = eagle_logits_2x[-labels.shape[1] :, :, :]
```
Comment (ChenhanYu) [inline modelopt/torch/speculative/plugins/megatron_eagle.py:1344]: So the range was wrong before?

**Review by ChenhanYu — COMMENTED** [review]
Left a question.

File: modelopt/torch/speculative/plugins/megatron_eagle.py (hunk 1 above)
Comment (yeyu-nvidia) [reply modelopt/torch/speculative/plugins/megatron_eagle.py:1344]: Not really. eagle_logits_2x has shape [2s,1, h]. So [-s:] is equivalent to [s:]. I just want it to be consistent to 3x and 4x

Comment (ChenhanYu) [reply modelopt/torch/speculative/plugins/megatron_eagle.py:1344]: OK
